corebehrt/common/setup.py

```diff
@@ -27,191 +32,360 @@ def get_args(default_config_name, default_run_name=None):
-        if output_path is None:
-            raise ValueError(
-                "Either pretrain_model_path, model_path, or save_folder_path must be provided."
-            )
-        cfg.paths.output_path = output_path
-        cfg.paths.run_name = DirectoryPreparer.construct_finetune_model_dir_name(cfg)
-        return cfg
+        return uuid.uuid4().hex
 
-    @staticmethod
-    def construct_finetune_model_dir_name(cfg: Config) -> str:
+    def generate_finetune_run_id(self) -> str:
         """
         Constructs the name of the finetune model directory.
         Based on the outcome type, the censor type, and the number of hours pre- or post- outcome.
         """
-        outcome_name = DirectoryPreparer.get_event_name(cfg.paths.outcome)
+        suffix = (
+            self.cfg.paths.run_name
+            if hasattr(self.cfg.paths, "run_name")
+            else uuid.uuid4().hex
+        )
+
+        outcome_name = self.get_event_name(self.cfg.paths.outcome)
         censor_name = (
-            DirectoryPreparer.get_event_name(cfg.paths.exposure)
-            if cfg.paths.get("exposure", False)
+            self.get_event_name(self.cfg.paths.exposure)
+            if self.cfg.paths.get("exposure", False)
             else outcome_name
         )
-        finetune_folder_name = f"finetune_{outcome_name}_censored_"
-
-        n_hours_censor = cfg.outcome.get("n_hours_censoring", None)
+        n_hours_censor = self.cfg.outcome.get("n_hours_censoring", None)
         n_hours_str = (
             DirectoryPreparer.handle_n_hours(n_hours_censor)
             if n_hours_censor is not None
             else "at"
         )
+        if self.cfg.outcome.get("index_date", None) is not None:
+            censor_name = DirectoryPreparer.handle_index_date(
+                self.cfg.outcome.index_date
+            )
 
-        if cfg.outcome.get("index_date", None) is not None:
-            censor_name = DirectoryPreparer.handle_index_date(cfg.outcome.index_date)
-
-        finetune_folder_name = f"{finetune_folder_name}{n_hours_str}_{censor_name}"
+        run_id = f"finetune_{outcome_name}_censored_{n_hours_str}_{censor_name}"
 
-        n_hours_start_follow_up = cfg.outcome.get("n_hours_follow_up", None)
+        n_hours_start_follow_up = self.cfg.outcome.get("n_hours_follow_up", None)
         n_hours_follow_up_str = (
             DirectoryPreparer.handle_n_hours(n_hours_start_follow_up)
             if n_hours_start_follow_up is not None
             else "at"
         )
 
-        finetune_folder_name = f"{finetune_folder_name}_followup_start_{n_hours_follow_up_str}_index_date_{cfg.paths.run_name}"
-        return finetune_folder_name
+        return f"{run_id}_followup_start_{n_hours_follow_up_str}_index_date_{suffix}"
 
     @staticmethod
     def get_event_name(path: str) -> str:
+        """
+        Gets the event name from the path to the outcome file.
+        """
         return split(path)[-1].strip(".csv")
```

Comment:
It'd be better to use splitext from os.path

Reply:
Going forward, I would prefer using `pathlib` rather than strings in general. Could leave this for future PR

Reply:
I thought about this, it should not cause a conflict as it is for the sub command - but I think the change is good for clarity anyways. I will change it

Reply:
I agree on both (as I do not want to go through all paths with pathlib in this PR).